incubator/java-spring-boot2/image/project/Dockerfile

```diff
@@ -1,23 +1,50 @@
-FROM {{.stack.image.namespace}}/{{.stack.id}}:{{.stack.semver.major}}.{{.stack.semver.minor}} as compile
+FROM maven:3.6-ibmjava-8 as compile
+
+# Ensure up to date / patched OS
+RUN  apt-get -qq update \
+  && apt-get -qq install -y curl wget xmlstarlet unzip \
+  && DEBIAN_FRONTEND=noninteractive apt-get -qq upgrade -y \
+  && apt-get -qq clean \
+  && rm -rf /tmp/* /var/lib/apt/lists/*
+
+RUN  groupadd --gid 1000 java_group \
+  && useradd --uid 1000 --gid java_group --shell /bin/bash --create-home java_user \
+  && mkdir -p /mvn/repository \
+  && chown -R java_user:java_group /mvn
 
 #setup project folder for java build step
-COPY . /project
+USER java_user
+
+# Layer caching for stack dependencies
+LABEL appsody.stack={{.stack.image.namespace}}/{{.stack.id}}:{{.stack.semver.major}}.{{.stack.semver.minor}}.{{.stack.semver.patch}}
```

Comment:
I wasn't sure it would at that stage. It gets fuzzy.

Reply:
I changed (but left) the label only for the intermediate/compile stage image, which the CLI does not directly create (which means I don't believe it can otherwise apply a label)